### ISSUE TITLE
Implement dual index barcodes, delimited by `+`.

### ISSUE THREAD

**Issue by theJasonFan** [Singular-Genomics/singular-demux#60]

**Comment by jiangweiyao** [comment]
Is the "+" in the metrics file? Here is a sample "per_sample_metrics.tsc" I generated recently. We would want "TAAGACCCTACTGGGACATATTGA" to be "TAAGACCCTACT+GGGACATATTGA"

```
barcode_name	library_name	barcode	templates	perfect_matches	one_mismatch_matches	q20_bases	q30_bases	total_number_of_bases	fraction_matches	ratio_this_barcode_to_best_barcode	frac_q20_bases	frac_q30_bases	mean_index_base_quality
IDTprimer1	IDTprimer1	TAAGACCCTACTGGGACATATTGA	22515511	21976946	415353	6412264359	6080909697	6754653300	0.1774624689780733	1.0	0.949310656551388	0.900254894947754	35.9763337113098
```

**Comment by jiangweiyao** [comment]
I think the "+" is in the output FASTQ files. We would like them to be in the per_sample_metrics files as well. Maybe even in the most_frequent_unmatched.tsv and sample_barcode_hop_metrics.tsv if feasible.

**Comment by theJasonFan** [comment]
Looks like a bug that needs to be tracked down. Can you share a minimal example to reproduce the issue?

**Comment by jiangweiyao** [comment]
No. The forward and reverse barcodes are in separate columns. They are concatenated together for matching, but we want to have the "+" in there for reports. Let me pull some data.

**Comment by theJasonFan** [comment]
Note: updated issue description to reflect changes in two source code areas. One for sample metrics and one for "hop" metrics.
